WordPress/gutenberg
# Extract the Router APIs and context into a dedicated package

Pre-requisite for #49956

## What?

This PR extracts the router provider and some related APIs (useHistory, useLocation) to a dedicated package in order to unblock #49956 allowing us to access the router/history from commands that are independent of the site editor.

## How?

This is the exact same code and APIs we already have in the site editor. These APIs were not necessarily meant to be external APIs yet (even if I can see plugins wanting to actually use them). They are very naive, so I'm making all of them as private APIs for now.

## Testing Instructions

Nothing really to test, just make sure you can open and navigate the site editor.
